h5io/_h5io.py

```diff
@@ -123,19 +145,34 @@ def _triage_write(key, value, root, comp_kw, where):
         _create_titled_dataset(root, key, 'ndarray', value)
     elif sparse is not None and isinstance(value, sparse.csc_matrix):
         sub_root = _create_titled_group(root, key, 'csc_matrix')
-        _triage_write('data', value.data, sub_root, comp_kw,
-                      where + '.csc_matrix_data')
-        _triage_write('indices', value.indices, sub_root, comp_kw,
-                      where + '.csc_matrix_indices')
-        _triage_write('indptr', value.indptr, sub_root, comp_kw,
-                      where + '.csc_matrix_indptr')
+        cleanup_data = _triage_write(
+            'data', value.data, sub_root, comp_kw,
+            where + '.csc_matrix_data', cleanup_data=cleanup_data)
+        cleanup_data = _triage_write(
+            'indices', value.indices, sub_root, comp_kw,
+            where + '.csc_matrix_indices', cleanup_data=cleanup_data)
+        cleanup_data = _triage_write(
+            'indptr', value.indptr, sub_root, comp_kw,
+            where + '.csc_matrix_indptr', cleanup_data=cleanup_data)
     else:
-        raise TypeError('unsupported type %s (in %s)' % (type(value), where))
-
+        try:
+            from pandas import DataFrame, Series
+            if isinstance(value, (DataFrame, Series)):
+                if isinstance(value, DataFrame):
+                    title = 'pd_dataframe'
+                else:
+                    title = 'pd_series'
+                rootname = root.name
+                cleanup_data.append({title: (rootname, key, value)})
+        except ImportError:
```

Comment:
It can go next to the equivalent thing for `scipy`:

https://github.com/h5io/h5io/blob/master/h5io/_h5io.py#L12